### DISCUSSION TITLE
Some Code Style Make the code more and more clearly

### DISCUSSION THREAD

**Discussion by sgf** [dotnet/csharplang#1071] [General]
```
if( s?)    //if (s!=null)

if (s in {"a","aaa","bbb"})    //if(string[]{"a","aaa","bbb"}.contains(s))

for(d=1..100)    //for(int d=1;d<=100;d++;)

for(f=1.0f..100)    //for(float f=1.0f;f<100.0;f=f+0.1f )


if(var size=sizeof(structA);size>12)
{...}


```
these just little some ways maybe will Make the C# Code clearly

**Just discuss whether they will be adopted**

**welcome to Add your ideas.**

**Comment by Logerfo** [comment]
See https://github.com/dotnet/csharplang/issues/420

**Comment by sharwell** [comment]
@sgf It tends to work well to separate ideas into one idea per issue, since GitHub only allows one discussion thread per issue.

> `if( s?)`

Duplicate of #196 and/or #545, and later #833.

> `if (s in {"a","aaa","bbb"})`

Duplicate of #420.

> `for(d=1..100)`
> `for(f=1.0f..100)`

Essentially a duplicate of #198.

**Comment by Joe4evr** [comment]
> `if( s?)    //if (s!=null)`

This one is gonna be `if (s is {} nonNullS)` as per [this comment by Gafter](https://github.com/dotnet/csharplang/issues/306#issuecomment-335948566).

**Comment by bondsbw** [comment]
@Joe4evr 
>`if (s is {} nonNullS)`

Meh, that is terribly unreadable.  I'll rarely if ever use it.

But I'd probably use this regularly:
```c#
if? (expression) ...
```
`if?` checks whether `expression` is null.  (Like `if`, it would accept an `else` clause.)

I chose this syntax over the proposal because `if (s?)` leads to ambiguities with the null-conditional operator, as [explained by @HaloFour in #833](https://github.com/dotnet/csharplang/issues/833#issuecomment-323554157).

**Comment by orthoxerox** [comment]
`if (s!=null)` is probably one construct that I've never wanted to replace with some sugar. It's easy to write and easy to read, its meaning is obvious.

**Comment by sgf** [comment]
@orthoxerox 
but when some time,
```
if (SomeLongNameVars!=null && someDictionary[keyworld]!=null&& someDataRow[Colume]!=null)
//
if (SomeLongNameVars? && someDictionary[keyworld]? && someDataRow[Colume]?)
```


thats will be little better

**if the same code in linq or  lambda,thats will be more usefully**

**Comment by sgf** [comment]
i have another idea like this.
most time we use some code like this:

```
if(someDictionary[keyworld]==null? ){someDictionary[keyworld]=11 }
//can we  write like this:
someDictionary[keyworld]?=11;
```

**Comment by aluanhaddad** [comment]
I really like
```cs
if (s is null)
```

**Comment by ufcpp** [comment]
`s?` is probably inspired by Swift's `let s?` pattern. However, my first impression for `s?` is `s == null` because `?` means **nullable**. It is not intuitive that `s?` means **not-null**.

**Comment by sgf** [comment]
```
if(var size=callMethod;size>12)
{...}
```
this style is good,and the golang,C++ has supported. maybe python too;
